script_parameters.yml

```diff
@@ -0,0 +1,249 @@
+        params:
+          # Optionally, you can specify the parameter values in the `params` list
+          # instead of in the `arguments` hash, by using the `default` key.  But
+          # this does not work for credential parameters.
+          - name: my_boolean
+            type: bool
+            default: true
+          - name: my_integer
+            type: integer
+            default: 1234
+          - name: my_float
+            type: float
+            default: 0.1234
+          - name: my_string
+            type: string
+            default: this is a string
+          - name: my_file
+            type: file
+            default: 0  # ID of the file
+          - name: my_custom_credential
+            type: credential_custom
+          - name: my_aws_credential
+            type: credential_aws
+          - name: my_db_credential
+            type: credential_redshift
+        arguments:
+          my_custom_credential: 0  # ID of the credential
+          my_aws_credential: 0
+          my_db_credential: 0
+        source: |
+          # Parameter values are added to your container as environment variables.
+          # See the script params reference for a full list.
+
+          import os
+
+          # Type conversions are necessary.  Unfortunately, `bool()` does not work here.
+          my_boolean = os.environ['MY_BOOLEAN'] == 'true'
+          my_integer = int(os.environ['MY_INTEGER'])
+          my_float = float(os.environ['MY_FLOAT'])
+
+          my_string = os.environ['MY_STRING']
+          my_file_url = os.environ['MY_FILE_URL']
+
+          my_custom_credential = (os.environ['MY_CUSTOM_CREDENTIAL_USERNAME'], os.environ['MY_CUSTOM_CREDENTIAL_PASSWORD'])
+          my_aws_credential = (os.environ['MY_AWS_CREDENTIAL_ACCESS_KEY_ID'], os.environ['MY_AWS_CREDENTIAL_SECRET_ACCESS_KEY'])
+          my_db_credential = (os.environ['MY_DB_CREDENTIAL_USERNAME'], os.environ['MY_DB_CREDENTIAL_PASSWORD'])
+      on-success:
+        - r_script
+
+    r_script:
+      action: civis.scripts.r
+      input:
+        required_resources:
+          cpu: 256
+          memory: 512
+          disk_space: 1
+        params:
+          - name: my_boolean
+            type: bool
+          - name: my_integer
+            type: integer
+          - name: my_float
+            type: float
+          - name: my_string
+            type: string
+          - name: my_file
+            type: file
+          - name: my_custom_credential
+            type: credential_custom
+          - name: my_aws_credential
+            type: credential_aws
+          - name: my_db_credential
+            type: credential_redshift
+        arguments:
+          my_boolean: true
+          my_integer: 1234
+          my_float: 0.1234
+          my_string: this is a string
+          my_file: 0               # ID of the file
+          my_custom_credential: 0  # ID of the credential
+          my_aws_credential: 0
+          my_db_credential: 0
+        source: |
```

Comment:
Is this how you might write the R code?  I haven't really written in R before.

Reply:
Looks good